### PR TITLE
MMP-154, refactor validationMojo and create a java agnostic API

### PR DESCRIPTION
**Pull request by damiansima**
MMP-154, refactor initializeMojo

MMP-154,add thest for projectfoldersGenerator

MMP-154, final refactor and test for contentGenerator. Move classes to api package and minor fix on validations

MMP-154, change api to work with paths

MMP-154,remove usless class

MMP-154, minor refactor, remove unnecessary code, and improve tests

### CODE REVIEW

File: src/test/java/integrationTests/mojo/ValidateMojoTest.java [hunk 1]
```diff
@@ -58,38 +59,29 @@ public void testFailWhenMissingPackageType() throws Exception {
...
   public void testFailOnEmptyPolicyProject() throws Exception {
     projectBaseDirectory = builder.createProjectBaseDir(EMPTY_POLICY_NAME, this.getClass());
     verifier = buildVerifier(projectBaseDirectory);
-    try {
-      verifier.executeGoal(VALIDATE);
-    } catch (VerificationException e) {
-    }
-    verifier.verifyTextInLog("Invalid Mule project. Missing src/main/policy folder. This folder is mandatory");
+    verifier.executeGoal(VALIDATE);
+    // verifier.verifyTextInLog("Invalid Mule project. Missing src/main/policy folder. This folder is mandatory");
```
Comment (WilliamPinho) [inline src/test/java/integrationTests/mojo/ValidateMojoTest.java]: Would it be possible to reactivate the commented verifications?

File: src/test/java/org/mule/tools/api/packager/ContentGeneratorTest.java [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+    PackagerTestUtils.createEmptyFolder(destinationFolderPath);
+
+    contentGenerator.createSrcFolderContent();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void createSrcFolderContentNonExistingDestinationFolder() throws IOException {
+    String sourceFolderName = packagingType.getSourceFolderName();
+
+    String fakeFileName = "fakeFile.xml";
```
Comment (WilliamPinho) [inline src/test/java/org/mule/tools/api/packager/ContentGeneratorTest.java]: Would it be possible to change it to constants?

File: src/test/java/org/mule/tools/api/packager/ContentGeneratorTest.java [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+            .resolve(GROUP_ID).resolve(ARTIFACT_ID);
+    PackagerTestUtils.createEmptyFolder(pomPropertiesDestinationPath);
+
+    contentGenerator.createDescriptors();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void createDescriptorsNoDescriptorDestinationFolder() throws IOException {
+    String pomFileName = "pom.xml";
+    String descriptorFileName = "mule-application.json";
```
Comment (WilliamPinho) [inline src/test/java/org/mule/tools/api/packager/ContentGeneratorTest.java]: Would it be possible to change these two to constants as well?

File: src/test/java/org/mule/tools/artifact/archiver/internal/PackageBuilderTest.java [hunk 4]
```diff
@@ -197,94 +206,94 @@ public void createDeployableFileSettingPluginsFolderTest() throws IOException {
...
+  // File notADirectory = mock(File.class);
+  //
+  // this.packageBuilder.generateArtifact(notADirectory, destinationFileMock);
+  // }
+  //
+  // @Test(expected = IllegalArgumentException.class)
+  // public void createArtifactInexistentDirectory() throws IOException {
+  // File inexistentDirectory = mock(File.class);
+  // when(inexistentDirectory.exists()).thenReturn(false);
+  //
```
Comment (WilliamPinho) [inline src/test/java/org/mule/tools/artifact/archiver/internal/PackageBuilderTest.java]: Please reactivate these tests?

File: src/test/java/org/mule/tools/api/packager/ValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    validator.isProjectStructureValid(MULE_POLICY);
+  }
+
+  @Test
+  public void isProjectStructureValidMulePolicy() throws ValidationException {
+    Path muleMainSrcFolder =
+        projectBaseFolder.getRoot().toPath().resolve(SRC.value()).resolve(MAIN.value()).resolve(POLICY.value());
+    muleMainSrcFolder.toFile().mkdirs();
+
+    Boolean valid = validator.isProjectStructureValid(MULE_POLICY);
```
Comment (WilliamPinho) [inline src/test/java/org/mule/tools/api/packager/ValidatorTest.java]: Would it be possible to check the case where the project that is being validated has a valid structure of another packaging type (for instance MULE_APPLICATION)?

File: src/test/java/org/mule/tools/api/packager/ContentGeneratorTest.java (hunk 2 above)
Comment (damiansima) [inline src/test/java/org/mule/tools/api/packager/ContentGeneratorTest.java]: done

File: src/test/java/org/mule/tools/api/packager/ContentGeneratorTest.java (hunk 3 above)
Comment (damiansima) [inline src/test/java/org/mule/tools/api/packager/ContentGeneratorTest.java]: DONE

File: src/test/java/org/mule/tools/api/packager/ValidatorTest.java (hunk 5 above)
Comment (damiansima) [inline src/test/java/org/mule/tools/api/packager/ValidatorTest.java]: done

File: src/test/java/org/mule/tools/artifact/archiver/internal/PackageBuilderTest.java (hunk 4 above)
Comment (damiansima) [inline src/test/java/org/mule/tools/artifact/archiver/internal/PackageBuilderTest.java]: done, I ended up removing them as they used an already deleted method and they validated logic of and old phase

File: src/test/java/integrationTests/mojo/ValidateMojoTest.java (hunk 1 above)
Comment (damiansima) [inline src/test/java/integrationTests/mojo/ValidateMojoTest.java]: DONE, I've re enabled all of them that are not path dependent.

**Pull request merged**